src/platform/nrfconnect/wifi/NrfWiFiDriver.cpp

```diff
@@ -206,77 +213,94 @@ void NrfWiFiDriver::ConnectNetwork(ByteSpan networkId, ConnectCallback * callbac
     WiFiManager::Instance().Connect(mStagingNetwork.GetSsidSpan(), mStagingNetwork.GetPassSpan(), handling);
 
 exit:
-    if (status != Status::kSuccess)
+    if (Instance().mpConnectCallback)
```

Comment:
`Instance()` is unnecessary. Also, shouldn't we only call it on failure? Otherwise, `Connect` is responsible for calling the callback, no?

Reply:
I can't see any "return" after `Connect`, so that didn't seem to be the case.

Reply:
Ahh right, thanks for the catch :)